### PR TITLE
Fancybox for group-picture and post-gallery

### PR DESCRIPTION
**Pull request by stevenjoezhang**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes was maked (for bug fixes / features).
   - [x] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [ ] Docs in [NexT website](https://theme-next.org/docs/) have been added / updated (for new features).

## PR Type
**What kind of change does this PR introduce?**

- [ ] Bugfix.
- [x] Feature.
- [ ] Code style update (formatting, local variables).
- [x] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?

Issue resolved: #296
See: https://github.com/iissnan/hexo-theme-next/commit/60778779273292f135a7c18360a2012ae0bc2a58

## What is the new behavior?

- Screenshots with this changes: N/A
- Link to demo site with this changes: N/A

### How to use?
In NexT `_config.yml`:
```yml
...
```

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

**Comment by 1v9** [comment]
screenshot? never use this feature...😂

**Comment by stevenjoezhang** [comment]
@1v9 See: #296

**Comment by 1v9** [comment]
@stevenjoezhang I tested group pictures, it's fancybox now, good job.

**Comment by stevenjoezhang** [comment]
You can also try this:
```
---
title: title
date: 2019-03-09 17:52:00
photos:
  - https://marketplace-images.githubusercontent.com/2388/1c118380-1ff1-11e9-89b2-1402fd4f7f2d
  - https://marketplace-images.githubusercontent.com/2388/1c118380-1ff1-11e9-89b2-1402fd4f7f2d
  - https://marketplace-images.githubusercontent.com/2388/1c118380-1ff1-11e9-89b2-1402fd4f7f2d
  - https://marketplace-images.githubusercontent.com/2388/1c118380-1ff1-11e9-89b2-1402fd4f7f2d
  - https://marketplace-images.githubusercontent.com/2388/1c118380-1ff1-11e9-89b2-1402fd4f7f2d
---

{% gp 5-2 %}
![](https://marketplace-images.githubusercontent.com/2388/1c118380-1ff1-11e9-89b2-1402fd4f7f2d)
![](https://marketplace-images.githubusercontent.com/2388/1c118380-1ff1-11e9-89b2-1402fd4f7f2d)
![](https://marketplace-images.githubusercontent.com/2388/1c118380-1ff1-11e9-89b2-1402fd4f7f2d)
![](https://marketplace-images.githubusercontent.com/2388/1c118380-1ff1-11e9-89b2-1402fd4f7f2d)
![](https://marketplace-images.githubusercontent.com/2388/1c118380-1ff1-11e9-89b2-1402fd4f7f2d)
{% endgp %}
```

File: layout/_macro/post.swig [hunk 1]
```diff
@@ -271,11 +271,7 @@
           {% set COLUMN_NUMBER = 3 %}
           {% for photo in post.photos %}
             {% if loop.index0 % COLUMN_NUMBER === 0 %}<div class="post-gallery-row">{% endif %}
-              <a class="post-gallery-img fancybox"
-                 href="{{ url_for(photo) }}" rel="gallery_{{ post._id }}"
-                 itemscope itemtype="http://schema.org/ImageObject" itemprop="url">
```
Comment (ivan-nginx) [inline layout/_macro/post.swig]: But why u remove `http://schema.org/ImageObject` here?

Comment (stevenjoezhang) [reply layout/_macro/post.swig]: This attribute belongs to the `<a>` tag which causes the bug, and that's why I removed it. I'll try to add it back, and test if fancybox still works fine.

**Comment by stevenjoezhang** [comment]
I make some other big changes to fit both fancybox 2 and 3, you can check again

**Comment by stevenjoezhang** [comment]
@maple3142 What happened?

**Comment by 1v9** [comment]
@stevenjoezhang answer is [here](https://github.com/maple3142/Blog/commits/master) 😂

**Comment by stevenjoezhang** [comment]
@1v9 Is it a mistake?

**Comment by 1v9** [comment]
@stevenjoezhang yeah i think so. merge upstream -> push to NexT?

**Comment by stevenjoezhang** [comment]
Done

**Comment by maple3142** [comment]
Sorry, I clone NexT theme from this repo to my site.
And I accidentally pushed it(theme-next/hexo-theme-next) instead of main repo(Blog) after I update it from upstream.

**Comment by 1v9** [comment]
@stevenjoezhang 👍
@maple3142 so dangerous😂

File: layout/_macro/post.swig [hunk 2]
```diff
@@ -271,11 +271,7 @@
...
-                <img src="{{ url_for(photo) }}" itemprop="contentUrl"/>
-              </a>
+              <img src="{{ url_for(photo) }}" itemprop="contentUrl"/>
```
Comment (ivan-nginx) [inline layout/_macro/post.swig]: Maybe we must add `itemscope itemtype="http://schema.org/ImageObject"` here, before `itemprop="contentUrl"`? It's for schema.

File: source/js/src/utils.js [hunk 3]
```diff
@@ -8,24 +8,30 @@ NexT.utils = NexT.$u = {
...
+        var imageTitle = $image.attr('title') || $image.attr('alt');
         var $imageWrapLink = $image.parent('a');
 
         if ($imageWrapLink.length < 1) {
-          var imageLink = $image.attr('data-original') ? this.getAttribute('data-original') : this.getAttribute('src');
-          $imageWrapLink = $image.wrap('<a data-fancybox="group" href="' + imageLink + '"></a>').parent('a');
-          $imageWrapLink.addClass('fancybox fancybox.image');
-          $imageWrapLink.attr('rel', 'group');
+          var imageLink = $image.attr('data-original') || $image.attr('src');
+          $imageWrapLink = $image.wrap('<a class="fancybox fancybox.image" href="' + imageLink + '" itemscope itemtype="http://schema.org/ImageObject" itemprop="url"></a>').parent('a');
```
Comment (stevenjoezhang) [inline source/js/src/utils.js]: @ivan-nginx Maybe the answer is no? See here.

Comment (ivan-nginx) [reply source/js/src/utils.js]: Oh, ok ok. So long line there...

**Pull request merged**